### PR TITLE
vote: encapsulate `Lockout`

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

lack of encapsulation in `vote_state::Lockout` clutters https://github.com/solana-labs/solana/pull/29524 with unnecessary plumbing

#### Summary of Changes

encapsulate it

this should be purely cosmetic aside from the note below

### CODE REVIEW

File: sdk/program/src/vote/state/mod.rs [hunk 1]
```diff
@@ -86,6 +90,18 @@ impl Lockout {
...
+    pub fn slot(&self) -> Slot {
+        self.slot
+    }
+
+    pub fn confirmation_count(&self) -> u32 {
+        self.confirmation_count
+    }
+
+    pub fn increase_confirmation_count(&mut self, by: u32) {
+        self.confirmation_count = self.confirmation_count.saturating_add(by)
```
Comment (t-nelson) [inline sdk/program/src/vote/state/mod.rs:103]: NOTE: switch to saturating from unchecked math here

Comment (t-nelson) [reply sdk/program/src/vote/state/mod.rs:103]: oh uhh... yeah looks like i missed a couple instances of same-module-pub.  will update. thanks!

File: sdk/program/src/vote/state/mod.rs [hunk 2]
```diff
@@ -544,8 +559,8 @@ impl VoteState {
         for (i, v) in self.votes.iter_mut().enumerate() {
             // Don't increase the lockout for this vote until we get more confirmations
             // than the max number of confirmations this vote has seen
-            if stack_depth > i + v.confirmation_count as usize {
-                v.confirmation_count += 1;
+            if stack_depth > i + v.confirmation_count() as usize {
+                v.increase_confirmation_count(1);
```
Comment (t-nelson) [inline sdk/program/src/vote/state/mod.rs:563]: got 'eem!

File: sdk/program/src/vote/state/mod.rs (hunk 1 above)
Comment (t-nelson) [reply sdk/program/src/vote/state/mod.rs:103]: https://github.com/solana-labs/solana/pull/29753#discussion_r1080664805

**Review by AshwinSekar — APPROVED** [review]
lgtm, thanks!

**Pull request merged**